### PR TITLE
Add DFU / BOOTLOADER support

### PR DESCRIPTION
**Pull request by AdamLaurie**
Hopefully this all makes sense! A lot of changes came together so some sanity checking probably required! :)

### CODE REVIEW

File: firmware/src/kb-rzusbstick/include/arch/avr/usb_descriptors.h [hunk 1]
```diff
@@ -130,8 +132,22 @@
...
+{ Usb_unicode('K') \
+ ,Usb_unicode('I') \
+ ,Usb_unicode('L') \
+ ,Usb_unicode('L') \
+ ,Usb_unicode('E') \
+ ,Usb_unicode('R') \
+ ,Usb_unicode('B') \
+ ,Usb_unicode('0') \
+ ,Usb_unicode('0') \
+ ,Usb_unicode('2') \
```
Comment (rmspeers) [inline firmware/src/kb-rzusbstick/include/arch/avr/usb_descriptors.h]: Should version numbering match the firmware name, e.g. 006?

File: killerbee/dev_rzusbstick.py [hunk 2]
```diff
@@ -245,48 +259,81 @@ def get_dev_info(self):
...
             except usb.USBError, e:
                 if e.args != ('No error',): # http://bugs.debian.org/476796
                     raise e
-            time.sleep(0.0005)
-            if response != RZ_RESP_SUCCESS:
-                if response in RESPONSE_MAP:
-                    raise Exception("Error: %s" % RESPONSE_MAP[response])
-                else:
-                    raise Exception("Unknown USB write error: 0x%02x" % response)
+#            time.sleep(0.0005)
```
Comment (rmspeers) [inline killerbee/dev_rzusbstick.py]: Great change to keep read centralized. We should remove these commented out lines to make it clean. I can do that in a separate PR afterwards though if easiest.

File: firmware/src/kb-rzusbstick/include/arch/avr/usb_descriptors.h (hunk 1 above)
Comment (AdamLaurie) [reply firmware/src/kb-rzusbstick/include/arch/avr/usb_descriptors.h]: Makes sense! I'll tweak...

File: killerbee/dev_rzusbstick.py (hunk 2 above)
Comment (AdamLaurie) [reply killerbee/dev_rzusbstick.py]: Yeah, I tend to leave them in particularly in other people's code so I'll let you decide what shold stay and what should go...

**Pull request merged**